src/server/game/Spells/Spell.cpp

```diff
@@ -3821,6 +3821,10 @@ void Spell::update(uint32 difftime)
             {
                 SendChannelUpdate(0);
                 finish();
+
+                if (Creature* creatureCaster = m_caster->ToCreature())
```

Comment:
random thought, why not inside Spell::finish() ?

Reply:
because there is no need for that anymore. spell_finish_successful cast already covers all cases where a spell is getting flagged as 'finished' straight afterwards. This case here is explicitely only for finished channel cases